src/Dotnet.Script.Core/ScriptPublisher.cs

```diff
@@ -97,6 +98,10 @@ public ScriptPublisher(LogFactory logFactory, ScriptEmitter scriptEmitter)
         {
             try
             {
+                var nugetFile = Path.Combine(outputDirectory, "NuGet.config");
```

Comment:
I think that we would be better off copying this file to the temp folder from which we actually do the restore. Today we copy local NuGet.Config file to the temp folder and do the restore. That works fine unless the you have a NuGet.Config file located in a parent folder relative to the script folder. If we could resolve the properties and write that NuGet.Config file to the temp folder, it would also work from the OmniSharp side of things. OmniSharp also does a potential restore through dotnet script to gather the compilation dependencies needed for language services and such. In addition we would like this to work with the REPL as well. Bottom line here is that this needs to be moved closer into the process of actually doing the underlying dotnet restore so that it works in all use cases and not only when publishing

Reply:
Which would mean that this code probably belongs in the `Dotnet.Script.DependencyModel` library. It also pulls in the `NuGet.Configuration` package . @filipw Is that going to give us any issues over at OmniSharp? OmniSharp already uses a handful of other NuGet related packages.

https://github.com/OmniSharp/omnisharp-roslyn/blob/db13beab49001ec95da7786e29a98ece60e064de/build/Packages.props#L58

Reply:
It makes sense, in that way all NuGet properties would be available even if the config file is in a prent folder.
It seems also to be very straightforward, as it can be done by only replacing `ScriptProjectProvider.CopyNuGetConfigFile` with the utility I added.

Reply:
Exactly 👍

Reply:
Do that and I'll have another look. 👍😎

Reply:
Nuget packages used to cause problems in OmniSharp since OmniSharp ships with an embedded Mono which trims all the unused types to eventually arrive at 5MB or so runtime. The old packages used drag in a ton of dependencies but since v4 this is no longer the case. I would only advise to use 4.8.0 to align exactly with the rest of O#